utils.js

```diff
@@ -0,0 +1,224 @@
+    };
+  } );
+}
+
+const getReactionRel = (content) => {
+  const reactionReactantRecords = [];
+  const reactionProductRecords = [];
+  const reactionGeneRecords = [];
+  const reactionSubsystemRecords = [];
+  content.reaction.forEach((r) => {
+    Object.entries(r.metabolites).forEach((e) => {
+      const [ compartmentalizedMetaboliteId, stoichiometry ] = e;
+      if (stoichiometry < 0) {
+        reactionReactantRecords.push({ compartmentalizedMetaboliteId, reactionId: r.reactionId, stoichiometry: -stoichiometry });
+      } else {
+        reactionProductRecords.push({ reactionId: r.reactionId, compartmentalizedMetaboliteId, stoichiometry });
+      }
+    });
+    getGeneIdsFromGeneRule(r.geneRule).forEach((geneId) => {
+      reactionGeneRecords.push({ reactionId: r.reactionId, geneId });
+    });
+    r.subsystems.forEach((name) => {
+      reactionSubsystemRecords.push({ reactionId: r.reactionId, subsystemId: idfyString(name) });
+    })
+  });
+  return [reactionReactantRecords, reactionProductRecords, reactionGeneRecords, reactionSubsystemRecords];
+}
+
+const getGeneIdsFromGeneRule = (geneRule) => {
+  let idList = [];
+  if (geneRule) {
+    idList = geneRule.split(/\s+and\s+|\s+or\s+/).filter(e => e);
+  }
+  return idList;
+}
+
+const getUniqueCompartmentlizedMap = (m, hm, uniqueCompartmentalizedMap) => {
+  const newID = idfyString2(m.name);
+  if (!(newID in hm)) {
+    hm[newID] = m.name;
+    uniqueCompartmentalizedMap[m.compartmentalizedMetaboliteId] = newID;
+  } else {
+    if (hm[newID] !== m.name) {
+      // console.log('Error duplicated ID:' + newID + '(' + m.name + ') collision with ' + hm[newID]);
+      uniqueCompartmentalizedMap[m.compartmentalizedMetaboliteId] = newID + '_';
+    } else {
+      uniqueCompartmentalizedMap[m.compartmentalizedMetaboliteId] = newID;
+    }
+  }
+}
+
+const getUniqueMetabolite = (m, uniqueCompartmentalizedMap, uniqueMetDict, uniqueMetabolites) => {
+  const newID = uniqueCompartmentalizedMap[m.compartmentalizedMetaboliteId];
+  if (!(m.name in uniqueMetDict)) {
+    const uMet = {
+      metaboliteId: newID,
+      name: m.name,
+      alternateName: m.alternateName,
+      synonyms: m.synonyms,
+      description: m.description,
+      formula: m.formula,
+      charge: m.charge,
+      isCurrency: m.isCurrency,
+    };
+    uniqueMetabolites.push(uMet);
+    uniqueMetDict[uMet.name] = uMet;
+  }
+}
+
+const getComponentIdDict = (content, isHuman) => {
+  const componentIdDict = {}; // store for each type of component the key  Id <-> element
+  // use to filter out annotation/external ids for components not in the model and to add missing information
+  // extracted from these annotation files such as description, etc...
+  Object.keys(content).forEach((k) => {
+    componentIdDict[k] = Object.fromEntries(content[k].map(e => [e[`${k}Id`], e]));
+  });
+
+  if (isHuman) {
+    Object.keys(componentIdDict.gene).forEach((geneId) => {
+      humanGeneIdSet.add(geneId);
+    });
+  }
```

Comment:
This is related to the other comment about `humanGeneIdSet` not being set problem.

Reply:
Hi @e0 and @inghylt, thanks for pointing out this bug. It has been addressed in the commit [c6ec8d9](https://github.com/MetabolicAtlas/neo4j-data-generation/pull/16/commits/c6ec8d99ca342735540ae4a91c3c1e3a2bbb6eb2)